thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java

```diff
@@ -291,6 +293,41 @@ protected Long loadValue() {
         return count();
       }
     });
+
+    metricRegistry.register("taskLatencyInMillis", new CachedGauge<Long>(15, TimeUnit.SECONDS) {
+      @Override
+      protected Long loadValue() {
+        final List<TaskStatus> pendingStatus = List.of(TaskStatus.WAITING, TaskStatus.RUNNING);
+        final Timestamp taskLatencyWindowEnd = new Timestamp(System.currentTimeMillis());
+        final DaoFilter filter = new DaoFilter()
+            .setPredicate(Predicate.OR(
+                // tasks which are still active in the lifecycle
+                Predicate.IN("status", pendingStatus.toArray()),
+                // tasks which completed(updated with the final status) lifecycle in the window
+                Predicate.AND(
+                    Predicate.GE("updateTime", taskLatencyWindowStart),
+                    Predicate.LT("updateTime", taskLatencyWindowEnd)
+                )
+            ));
+        final List<TaskDTO> tasks = filter(filter);
+        if(tasks.isEmpty()) {
+          return 0L;
+        }
+        final long maxLatency = tasks.stream().map(task -> {
```

Comment:
My understanding is that this was enough: 
```
select IFNULL(CURRENT_TIMESTAMP() - MIN(create_time), 0) 
from task_index 
where status in ('WAITING', 'RUNNING') and type = 'DETECTION';
```
--> 1 for DETECTION , 1 for NOTIFICATION maybe.

Am I missing cases?

Reply:
The tasks with fulfilled statuses such as ('COMPLETED', 'FAILED', 'TIMEOUT') , latency of these must also be considered from create_time to last status update_time (usually status change is the only update done on tasks)